src/core/mac/mac.cpp

```diff
@@ -172,22 +172,6 @@ void Mac::Scan(Operation aScanOperation, uint32_t aScanChannels, uint16_t aScanD
     StartOperation(aScanOperation);
 }
 
-bool Mac::IsActiveScanInProgress(void)
-{
-    return (mOperation == kOperationActiveScan) || (mPendingActiveScan);
-}
-
-bool Mac::IsEnergyScanInProgress(void)
-{
-    return (mOperation == kOperationEnergyScan) || (mPendingEnergyScan);
-}
-
-bool Mac::IsInTransmitState(void)
```

Comment:
This is fine, but we may actually un-inline this later (the data poll PR https://github.com/openthread/openthread/pull/3918 changes this to a `switch` actually). I think `switch` will help us not forget to add new cases as we add more tx related operation (e.g. indirect tx).